docs/SPATIAL_FILTERING.md

```diff
@@ -0,0 +1,109 @@
+
+#### Inline specification
+
+The inline specification for a spatial filter consists of the name of the CRS, followed by a semicolon, followed by a valid Polygon or Multipolygon encoded using WKT or hex-encoded WKB. It will look something like one or other of the following:
+
+- `EPSG:4326;POLYGON((...))` (WKT)
+- `EPSG:4269;01030000...` (hex-encoded WKB)
+
+#### File containing the specification
+
+The file should contain either the name of the CRS or the entire CRS definition in WKT, followed by a blank line, followed by a valid Polygon or Multipolygon encoded using WKT or hex-encoded WKB. For example:
+
+```
+EPSG:4326
+
+POLYGON((...))
+```
+
+To reference a spatial filter file on your filesystem, use an @ symbol followed by the path to the file. This syntax can be used instead of the inline specification anywhere that a spatial filter specification is expected.
+
+### Setting the spatial filter
+
+This can be done during a clone operation (which can save you from downloading unneeded data, see # During clone for details).
+- `kart clone URL --spatial-filter="EPSG:4326;POLYGON((...))"`
+
+Or to reference a file containing the spatial filter specification:
+- `kart clone URL --spatial-filter=@myspatialfilter.txt`
+
+
+To change the spatial filter at any time after the clone:
+- `kart checkout --spatial-filter="EPSG:4326;POLYGON((...))`
+- `kart checkout --spatial-filter=@myspatialfilter.txt`
+
+To clear the spatial filter (such that all features match the spatial filter, as happens by default):
+- `kart checkout --spatial-filter=none`
+
+The command `kart status` will tell you if a spatial filter is currently active.
+
+### Indexing
+
+Indexing is only useful if you are managing a repository will be cloned by others who will do so using a spatial filter. In this case, you should run the following command to generate a spatial index the repository so that when they clone it, they only receive the features that match the spatial filter. If there is no spatial index, they will instead receive every single feature. As explained above, the resulting repository will still behave as expected - Kart commands and working copy will still be limited precisely according to the spatial filter specified - but they will miss out on the saving of only downloading the required data.
+
+To index a repository, run the following command:
+
+`kart spatial-filter index`
+
+As more data is added to the repository, running the same command again will index data that has not yet been indexed. Running this command on a semi-regular basis
+as the repository has more data added will help ensure users get the most efficient spatially filtered clones possible, but forgetting to do so has no adverse effects apart from reduced efficiency. This could be automated by using, for instance, the [git post-receive hook](https://git-scm.com/docs/githooks#post-receive).
```

Comment:
```suggestion
As more data is added to the repository, running the same command again will index data that has not yet been indexed. Running this command on a semi-regular basis as the repository has more data added will help ensure users get the most efficient spatially filtered clones possible, but forgetting to do so has no adverse effects apart from reduced efficiency. This could be automated by using, for instance, the [git post-receive hook](https://git-scm.com/docs/githooks#post-receive).
```